src/plugins/controls/public/control_group/component/control_group_component.test.tsx

```diff
@@ -0,0 +1,197 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { stubDataView } from '@kbn/data-views-plugin/common/data_view.stub';
+import { pluginServices as presentationUtilPluginServices } from '@kbn/presentation-util-plugin/public/services';
+import { registry as presentationUtilServicesRegistry } from '@kbn/presentation-util-plugin/public/services/plugin_services.story';
+import { act, render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Provider } from 'react-redux';
+import { OptionsListEmbeddableFactory } from '../..';
+import { OPTIONS_LIST_CONTROL, RANGE_SLIDER_CONTROL } from '../../../common';
+import { mockControlGroupContainer, mockControlGroupInput } from '../../../common/mocks';
+import { RangeSliderEmbeddableFactory } from '../../range_slider';
+import { pluginServices } from '../../services';
+import { ControlGroupContainerContext } from '../embeddable/control_group_container';
+import { ControlGroupComponentState, ControlGroupInput } from '../types';
+import { ControlGroup } from './control_group_component';
+
+jest.mock('@dnd-kit/core', () => ({
+  /** DnD kit has a memory leak based on this layout measuring strategy on unmount; setting it to undefined prevents this */
+  ...jest.requireActual('@dnd-kit/core'),
+  LayoutMeasuringStrategy: { Always: undefined },
+}));
+
+describe('Control group component', () => {
+  interface MountOptions {
+    explicitInput?: Partial<ControlGroupInput>;
+    initialComponentState?: Partial<ControlGroupComponentState>;
+  }
+
+  presentationUtilServicesRegistry.start({});
+  presentationUtilPluginServices.setRegistry(presentationUtilServicesRegistry);
+
+  pluginServices.getServices().dataViews.get = jest.fn().mockResolvedValue(stubDataView);
+  pluginServices.getServices().dataViews.getIdsWithTitle = jest
+    .fn()
+    .mockResolvedValue([{ id: stubDataView.id, title: stubDataView.getIndexPattern() }]);
+  pluginServices.getServices().controls.getControlTypes = jest
+    .fn()
+    .mockReturnValue([OPTIONS_LIST_CONTROL, RANGE_SLIDER_CONTROL]);
+  pluginServices.getServices().controls.getControlFactory = jest
+    .fn()
+    .mockImplementation((type: string) => {
+      if (type === OPTIONS_LIST_CONTROL) return new OptionsListEmbeddableFactory();
+      if (type === RANGE_SLIDER_CONTROL) return new RangeSliderEmbeddableFactory();
+    });
+
+  async function mountComponent(options?: MountOptions) {
+    const controlGroupContainer = await mockControlGroupContainer(
+      mockControlGroupInput(options?.explicitInput),
+      options?.initialComponentState
+    );
+
+    const controlGroupComponent = render(
+      <Provider
+        // this store is ugly, but necessary because we are using controlGroupSelector rather than controlGroup.select
```

Comment:
When I tried to switch to `controlGroup.select` in the  `ControlGroup` + `SortableControlInner` components, this caused a race condition where, on dashboard navigation / drilldown, the `SortableControlInner` would receive the updated panel IDs **before** the `ControlGroup` did, which means the ID it would try to access would be from the **old** control group.

Not entirely sure why this was happening, but since this is all going to be rewritten as part of the embeddable refactor anyway, I figured it wasn't worth fighting for now - hence, the ugly mock store :laugh: